### ISSUE TITLE
thunderblot 3 usb no use

### ISSUE THREAD

**Issue by jiaoting** [zearp/Nucintosh#63]
I've tested that the USB hub of thunderbolt can't be used. I don't know why. My USB hub is Lenovo's 40ac0135cn

**Comment by jiaoting** [comment]
I've tested that the USB hub of thunderbolt can't be used. I don't know why. My USB hub is Lenovo's ThinkPad Thunderbolt 3 Dock

**Comment by zearp** [comment]
Any Thunderbolt device needs to be connected before turning on the machine. All ports should then work on the dock. I don't own that particular dock but have tested my CalDigit dock without issues as long as it connected before powering on. Also make sure Thunderbolt security is set to legacy in the bios.

Do other ports work on your dock? There is very little information for me to go by in your issue. Such as logs and details on what does work when connected, how its connect, what happens when it's connected (system information/IOreg dumps), what macOS version, etc etc. 

Just stating X doesn't work on Y doesn't;t give me anything to go by unless I happen to have the exact same device and I do have a lot of stuff but Thunderbolt devices are expensive so I only have one dock myself. I've had reports of Thunderbolt eGPU enclosures working and other docks working too so I assume Thunderbolt is working and any device adhering to the standard should work.

If everything works in Windows you could compare the hardware/devices between the two operating systems. Generate a report with something HWiNFO64 and compare that to what macOS shows in its system information app or IOreg. Also check if the firmware is up-to-date and compatible with the macOS version you are running. In a quick search I could find some reports of it working on Macs but those posts were several years old.

I don't think the issue is caused by anything in the EFI but would be interested to hear if other EFI's from other repo's do make the dock work completely. It's easy to test them and it would be easy for me to compare and know what to do to make it work if you happen to find an EFI that makes the dock function completely.

I'm closing this issue as I won't be able to reproduce since I don't own this dock but if you can provide more information maybe it can be made to work. I can't really provide more steps or detailed that this reply as my time is limited and I don't the issue section of the repo turning into me providing step by step guides on how to troubleshoot every device someone may want to use. Doing that would only be possible when issues are opened that provide a lot of detail and logs/etc.

tl;dr
Thunderbolt docks and devices I tested worked as intended given they were connected before powering up and with Thunderbolt security mode set to legacy in the bios.
